### DISCUSSION TITLE
Dialog: Separate backdrop and content into separate sibling components

### DISCUSSION THREAD

**Discussion by crisshaker** [ariakit/ariakit#2085] [General]
#### Please note: This duration of the animation has been increased so the behavior is easy to understand.
https://user-images.githubusercontent.com/36159205/204778448-6127f1ae-8986-4e56-875f-1d2091068db7.mov

#### What I'm trying to achieve:
I want to animate the backdrop **only** to slowly fade in. I **don't** want the same effect on the contents (the bottom sheet in this case)

#### The problem
From the video, it's apparent it's not possible. This is because the contents are rendered as children of the backdrop.

#### What I've tried
Currently I'm resorting to a trick to get around that. I'm rendering a div as part of the children of Dialog and styling it as the backdrop (with pointer-events: none). Though this sort of works, there are some slight layout issues happening because of the div I'm using as the backdrop. Also, this generally feels like a hack.

#### Possible solution
I believe this can easily be solved by separating the overlay from the content. This way, we can animate them independently from each other. I'm familiar with this approach having used the dialog component from radix-ui. https://www.radix-ui.com/docs/primitives/components/dialog

**Comment by diegohaz** [comment]
They were separate components in the first versions of the library, but we've changed it so the backdrop wraps the dialog because they share too much state (portal, modal, hideOnInteractOutside, and other props that are passed to the dialog).

> Currently I'm resorting to a trick to get around that. I'm rendering a div as part of the children of Dialog and styling it as the backdrop (with pointer-events: none). Though this sort of works, there are some slight layout issues happening because of the div I'm using as the backdrop. Also, this generally feels like a hack.

Could you put together a CodeSandbox with that? I have a feeling that the actual solution may follow that path.
